sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAdministrationAsyncClient.java

```diff
@@ -0,0 +1,485 @@
+     * using an unsecured attestation token.</p>
+     *
+     * @param attestationType Specifies the trusted execution environment to be used to validate the evidence.
+     * @param newAttestationPolicy Specifies the policy to be set on the instance.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response to an attestation policy operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<PolicyResult>> setAttestationPolicyWithResponse(AttestationType attestationType, String newAttestationPolicy) {
+        return withContext(context -> setAttestationPolicyWithResponse(attestationType, newAttestationPolicy, context));
+    }
+
+    /**
+     * Sets the current policy for an attestation type with an unsecured attestation policy.
+     * <p>Note that this API will only work on AAD mode attestation instances, because it sets the policy
+     * using an unsecured attestation token.</p>
+     *
+     * @param attestationType Specifies the trusted execution environment to be used to validate the evidence.
+     * @param newAttestationPolicy Specifies the policy to be set on the instance.
+     * @param context Context for this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response to an attestation policy operation.
+     */
+    Mono<Response<PolicyResult>> setAttestationPolicyWithResponse(AttestationType attestationType, String newAttestationPolicy, Context context) {
+        AttestationPolicySetOptions options = new AttestationPolicySetOptions()
+            .setPolicy(newAttestationPolicy);
+        return setAttestationPolicyWithResponse(attestationType, options, context);
+    }
+//endregion
+
+    /**
+     * Sets the current policy for an attestation type.
+     *
+     * @param attestationType Specifies the trusted execution environment to be used to validate the evidence.
+     * @param options Options for the setPolicy operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response to an attestation policy operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<PolicyResult>> setAttestationPolicyWithResponse(AttestationType attestationType, AttestationPolicySetOptions options) {
```

Comment:
>setAttestationPolicyWithResponse [](http://example.com/codeflow?start=40&length=32)

You do not need to have multiple overloads for public withResponse method for options. We only do that for nonWithResponse method.


For example - you have 2 service methods for setAttestationPolicyWithResponse one that takes options parameter and another convenience overload that takes a string for attestationPolicy. You can only have 1 method with options for WithResponse instead. Have only the convenience overload for the setAttestionPolicy() method instead.